uaclient/security.py

```diff
@@ -753,45 +817,74 @@ def _prompt_for_attach(cfg: UAConfig) -> bool:
     return True
 
 
+def _check_subscription_for_required_service(
+    cfg: UAConfig, pocket: str
+) -> bool:
+    """Verify if the ua subscription has the required service enabled."""
+    if pocket == UA_INFRA_POCKET:
+        service_to_check = "esm-infra"
+    else:
```

Comment:
We probably want to specificatlly check elif pocket == UA_APPS_POCKET   and fall through to an else: for unknown pocket sources. It's likely that we have some pocket metadata gaps that could return None or unexpected pockets, and we want to warn in that case and return False or raise a  exceptions.SecurityAPIMetadataError.

Reply:
@blackboxsw I don't think we can reach a scenario like this, because of the way we retrieve the pocket information
from the `CVEPackageStatus` object. We use the `pocket_source` property:

https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/security.py#L224

You can see that there is currently no way for us to return something that is not one of:

* Ubuntu Standard Updates
* ESM Infra
* ESM Apps

However, it seems that the security api is returning the pocket information already. Should we update that property as well ?

Reply:
@lucasmoura I was mostly concerned with the fact that we prefer USN.release_packages.*pocket values if present before falling back to CVE.packages.statuses.*pocket values. So if USN has a unexpected value in pocket, we wouldn't handle that case here. But, you corrected me that we still call[ CVEPackageStatus.pocket_source](https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/security.py#L232)  which will properly put things into one of our known expected pockets in the "else" clause. We are still exposed to our potentially incorrect behavior of assuming that an unknown pocket means either "UA Infra" or "Ubuntu Standard Updates", but I think we can handle that in GH: #1376.